src/PaloCleaner.py

```diff
@@ -1683,151 +1700,234 @@ def replace_object_in_groups(self, location_name: str, progress: rich.progress.P
-                        elif source_obj_instance.about()['name'] in checked_object.value:
-                            matched = True
-
-                        # If the current object to be replaced has been matched as a member of a static group at the
-                        # current location level, add it to the replacements_done tracking dict
-                        if matched:
+            # TODO CHECK ASYNCIO?     
+                    
+        progress.update(task, advance=1)
+
+        def replace_objects_in_service_groups(replacement, replacement_name):
+            """
+            This function replaces the services objects for which a better duplicate has been found on the current location groups
+
+            :param replacement_name: (str) 
+            :param replacement: ()
+            :return:
+            """
+
+            while True:
+                if self._nb_thread:
+                    if self._queue.empty(): #TODO MANAGE EXCPETION
+                        break
+                    else:
+                        replacement_name, replacement = self._queue.get()
+
+
+                # the source object name is the key on the _replacements dict
+                source_obj = replacement_name
+                # the source_obj_instance and source_obj_location are found in the 'source' key of the dict item
+                source_obj_instance, source_obj_location = replacement['source']
+                # the replacement_obj_instance and replacement_obj_location are found in the 'replacement' key of the dict item
+                replacement_obj_instance, replacement_obj_location = replacement['replacement']
+
+                # for each ServiceObject type object in the current location objects
+                for checked_object in self._objects[location_name]['Service']:
+                    # if the type of the current object is a ServiceGroup
+                    if type(checked_object) is panos.objects.ServiceGroup and checked_object.value:
+                        changed = False
+                        matched = False
+                        try:
+                            # if the name of the replacement object is different than the origin one, then the static
+                            # group members values needs to be updated
+                            if source_obj_instance.about()['name'] != replacement_obj_instance.about()['name']:
+                                checked_object.value.remove(source_obj_instance.about()['name'])
+                                # adding the replacement object to the group only if it has not been already used to replace
+                                # another one (case where we have duplicate objects on a static group)
+                                if not replacement_obj_instance.about()['name'] in checked_object.value:
+                                    checked_object.value.append(replacement_obj_instance.about()['name'])
```

Comment:
Can lead to race condition if 2 objects in a group are replaced by the same one, if another thread is doing that at the same time (replacement object could be added 2 times to the list).